src/test/scala/com/snowplowanalytics/kinesistee/JavascriptOperatorSpec.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2016 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.kinesistee
+
+
+import com.snowplowanalytics.kinesistee.models.{FilteredContent, NonEmptyContent}
+import org.specs2.mutable.Specification
+import org.specs2.scalaz.ValidationMatchers
+
+import scalaz.syntax.validation._
+import scalaz.{Failure, Success}
+
+class JavascriptOperatorSpec extends Specification with ValidationMatchers {
+
+  "A valid JS Operator" should {
+
+    val jsTransform =
+      """
+        | function operator(row) {
+        |     return row.replace("$", "");
+        | }
+      """.stripMargin
+
+    val jsFilterTrue =
+      """
+        | function operator(row) {
```

Comment:
Agreed, my thoughts were to treat them as a single operator to avoid code duplication. The Javascript filter and operator are identical, besides their signature.